module5/lessons/Week1/TalkingWhileCoding.md

```diff
@@ -0,0 +1,150 @@
+
+### What is the purpose of a live technical coding challenge interview?
+
+<section class="call-to-action" markdown="1">
+With your partner: Brainstorm what competencies the interviewer is evaluating in a live technical coding challenge.
+</section>
+
+<aside class="instructor-notes" markdown="1">
+    <p><strong>Instructor Note</strong><br>
+    Take notes of what students share in Slack so they can reference it later.
+    We're specifically looking for problem-solving ability, ability to work through a challenge when they don't know what to do next, and communication/collaboration skills. In addition to technical proficiency. How would this person be to work with?</p>
+</aside>
+
+
+### What To Say
+
+You've heard that you should be talking pretty much the entire time while working on a live technical coding challenge, and you may be wondering what sort of things you should say.
+
+A Few Ideas:
+* Repeat back the question to make sure you understand
+* Asking clarifying questions
+* Checking in to see if your interviewer thinks you are on the right track
+* Sharing your high-level plan
+* Talking through design decisions, even very small ones
+* When you're stuck, explain what's tripping you up and what options you are considering
+
+### Why is Talking While Coding Important?
+
+<section class="call-to-action" markdown="1">
+With your Partner: Look back at the competencies your interviewer is evaluating that we listed earlier. Brainstorm why talking while you solve the problem would help you get a higher score on those competencies.
+</section>
+
+### Four Quick Tips
+
+### Tip 1. Focus on the Purpose
+
+When folks first start trying to talk in a technical interview, they sometimes start by literally reading their code out loud or talking in a way that sounds like reading code. This is not adding very much value, the interviewer can see what you're typing! Instead, focus on the purpose of your code.
+
+### Tip 2. Share Tradeoffs or Uncertainty
+
+Any time that you are considering different options, it's great to share that with your interviewer! Are you uncertain about the approach you are taking because you see a potential issue, that's useful to share too!
+
+### Tip 3. Think About Your Tone
+
+Try to maintain an enthusiastic approach, even when feeling nervous. You can to to come across as competent and humble, so try to find the middle ground between overconfidence and excessive uncertainty.
+
+### Tip 4. Take a Quiet Pause
+
+Yes, sharing your thought process is important, but if taking a minute to think quietly would help you gather your thoughts that's great too. Just make sure to communicate that with your interviewer instead of just going silent.
+
+
+### Let's Watch an Excellent Example
+
+Individually, make you way to [this](https://www.youtube.com/watch?v=1qw5ITr3k9E) mock technical interview. Watch from 3:30 through 18:08 when she finishes the `Book` class.

```

Comment:
This is a great interview example!

Reply:
I agree! Was very happy when I found it! And I like the difficulty level of the problem too. I think her type of response is absolutely something our students can all shoot for.